File: frontend/hc_professional/src/api_types/Web_ReconcileAnomalyPayload.ts
/**
 * Sent by a healthcare provider or BHW to reconcile one or more anomalous
 * entries (e.g. mark a technical-miss as provider-verified).
 */
export interface WebReconcileAnomalyPayload {
    entry_ids:           number[];
    reason:              string;
    verification_method: AdherenceStatusEnum;
}

export enum AdherenceStatusEnum {
    AppRecorded = "app_recorded",
    ProviderReconciled = "provider_reconciled",
    TechnicalMiss = "technical_miss",
    UnverifiedAbsence = "unverified_absence",
}

File: frontend/patient/app/src/main/java/com/pinghtdog/amping/api_schemas/Web_ReconcileAnomalyPayload.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json                       = Json { allowStructuredMapKeys = true }
// val webReconcileAnomalyPayload = json.parse(WebReconcileAnomalyPayload.serializer(), jsonString)

package com.pinghtdog.amping.api_schemas

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

/**
 * Sent by a healthcare provider or BHW to reconcile one or more anomalous
 * entries (e.g. mark a technical-miss as provider-verified).
 */
@Serializable
data class WebReconcileAnomalyPayload (
    @SerialName("entry_ids")
    val entryIDS: List<Long>,

    val reason: String,

    @SerialName("verification_method")
    val verificationMethod: String
)
